test/NuGet.Core.Tests/NuGet.Commands.Test/MSBuildProjectFactoryTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using NuGet.Commands;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.Commands.Test
+{
+    public class MSBuildProjectFactoryTests
+    {
+        [PlatformTheory(Platform.Windows)]
+        [InlineData("D:\\temp\\project1\\source.cs",                            "D:\\temp\\project1",                           "src\\project1\\source.cs")]
+        [InlineData("D:\\temp\\project1\\folder1\\source.cs",                   "D:\\temp\\project1",                           "src\\project1\\folder1\\source.cs")]
+        [InlineData("D:\\temp\\project1\\folder1\\folder2\\source.cs",          "D:\\temp\\project1",                           "src\\project1\\folder1\\folder2\\source.cs")]
+        [InlineData("D:\\temp\\source.cs",                                      "D:\\temp\\project1",                           "src\\project1\\source.cs")]
+        public void MSBuildProjectFactory_GetTargetPathForSourceFiles_Windows(string sourcePath, string projectDirectory, string expectedResult)
```

Comment:
You don't need `MSBuildProjectFactory` in the test names.  It's redundant with the class name.